### ISSUE TITLE
Move library to the root of the repository

### ISSUE THREAD

**Issue by per1234** [end2endzone/NonBlockingRTTTL#3]
GitHub automatically generates .zip files of the repository contents. Moving the library to the root of the repository would allow that file to be used for installation, rather than requiring this complex deployment process. This will allow users to follow the same installation process as they would use for any other Arduino library hosted on GitHub, rather than a non-standard process.

This repository structure is also required if you want to add your library to the [Arduino Library Manager](https://github.com/arduino/Arduino/wiki/Library-Manager-FAQ) index, which provides an even easier installation and update process for the users.

My proposed repository structure is:
```
NonBlockingRTTL (repository root)
|_library.properties
|_keywords.txt
|_examples
|_src
  |_NonBlockingRtttl.h
  |_NonBlockingRtttl.cpp
```
That is the Arduino Library Format 1.5. If you prefer the 1.0 format then the source files can be placed in the repository root instead of in the src subfolder.

**Comment by end2endzone** [comment]
Thank you for your feedback. Much appreciated. I didn't know about the Arduino Library Manager. I know my deployment process is complicated but didn't have a better alternative. I will make the corrections soon and open an issue on [Arduino's GitHub](https://github.com/arduino/Arduino/issues) for proposing the library.

**Comment by per1234** [comment]
Library Manager is really great because the Arduino IDE provides update notifications whenever a new version is released of any of the libraries in the Library Manager index you have installed. You can then update to the new version with only a few clicks. It makes it so much easier to keep things up to date. You can also use Library Manager to easily roll back to older versions if there is some incompatibility introduced by the current release.

I'd be happy to submit a pull request for the proposed folder structure and library.properties metadata file if that would be helpful. I had started to do this but then noticed you had the whole deployment system set up so I decided it would be better to open an issue to get an idea of how receptive you would be to it.

**Comment by end2endzone** [comment]
Yes,  feel free to send a pull request. That would be perfect.

> [...] but then noticed you had the whole deployment system set up

Are you referring to the `Installing` section of `INSTALL.md`? I don't care changing my current deployment setup. This will clear up any confusion between the `github autogenerated zip` and the `library installer`. Deploying through the Arduino Library Manager will be much easier and streamline the process. And as you said before, other Arduino libraries are already setup like that.

P.S. Don't forget to also add your referral to [AUTHORS](../blob/master/AUTHORS) file.

**Comment by end2endzone** [comment]
As a side note, if you are using the library, you should know that I am in the process of migrating all my arduino libraries from a personal web page to GitHub. In occurrence, if you need RTTTL playback support, you should also take a look at [AnyRtttl](http://www.end2endzone.com/anyrtttl-a-feature-rich-arduino-library-for-playing-rtttl-melodies/) which has more features than NonBlockingRtttl.

**Comment by end2endzone** [comment]
Fixed in e716f81b6a0d49268c57825ed63b0c6307ed2488

**Comment by end2endzone** [comment]
@per1234 A request was created on Arduino's GitHub to add NonBlockingRTTTL to Library Manager: 

[Please add NonBlockingRTTTL](https://github.com/arduino/Arduino/issues/7659)

Thanks again!
